lib/buffer.js

```diff
@@ -1259,7 +1259,28 @@ function atob(input) {
   if (arguments.length === 0) {
     throw new ERR_MISSING_ARGS('input');
   }
-  input = `${input}`;
+
+  if (input === undefined || input === false || typeof input === 'number') {
+    throw lazyDOMException(
+      'The string to be decoded is not correctly encoded.',
+      'ValidationError');
+  }
+
+  // Remove all ASCII whitespace from data.
+  //
+  // See #1 - https://infra.spec.whatwg.org/#forgiving-base64
+  input = `${input}`.replace(/\s/g, '');
```

Comment:
Is there a preferred approach for stripping out **all** whitespace characters? Any utilities that I should be aware of?

Reply:
Here required not all whitespaces - only ASCII - `/[\t\n\f\r ]+/g`.

Reply:
Another approach would be to count all non ASCII whitespace character in the loop below instead of modifying the string, we might get better performance out of it.

Reply:
Thanks for the great suggestions! @aduh95 I've adopted your approach of character counting in the latest commit.